kqueue.c

```diff
@@ -420,9 +384,9 @@ kq_dispatch(struct event_base *base, struct timeval *tv)
 	}
 
 	if (res == kqop->events_size) {
-		/* We used all the events space that we have. Maybe we should
-		   make it bigger. */
-		kq_grow_events(kqop, kqop->events_size * 2);
+		/* We've used up all the events space in this round.
+		 * Try to double the size of the eventlist for next round. */
+		kq_grow_events(kqop);
```

Comment:
I guess we can check exit code here as well

Reply:
Done.